popen3.c

```diff
@@ -0,0 +1,170 @@
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include "popen3.h"
+
+static void close_pipe(int fds[2])
+{
+	if(fds[0]) {
+		close(fds[0]);
+		fds[0] = -1;
+	}
+	if(fds[1]) {
```

Comment:
I believe this needs to be compared to -1, not 0.

Reply:
Right again. Corrected.